java/trace/src/main/scala/org/typelevel/otel4s/java/trace/TracerImpl.scala

```diff
@@ -68,14 +68,15 @@ private[java] class TracerImpl[F[_]: Sync: AskVault](
   def joinOrRoot[A, C: TextMapGetter](carrier: C)(fa: F[A]): F[A] = {
     val context = propagators.textMapPropagator.extract(Vault.empty, carrier)
 
+    // use Local[F, Vault].scope(..)(context) to bring extracted headers
     SpanContext.fromContext(context) match {
       case Some(parent) =>
-        childScope(parent)(fa)
+        Local[F, Vault].scope(childScope(parent)(fa))(context)
       case None =>
-        rootScope(fa)
+        Local[F, Vault].scope(fa)(context)
```

Comment:
Here is the actual fix

Reply:
Dam, this is what I was missing! Had done the same changes as you've done to the context conversions but still couldn't get it working. This is what I was missing. Nice one!